### PR TITLE
dos2unix: add autoupdate section

### PR DESCRIPTION
**Pull request by rasa**

### CODE REVIEW

File: bucket/dos2unix.json [hunk 1]
```diff
@@ -21,6 +21,16 @@
     ],
     "checkver": {
         "url": "https://waterlan.home.xs4all.nl/dos2unix.html",
-        "re": "Latest version: (?<version>[\\d.]+) \\([\\d\\-]+\\)"
+        "re": "Latest version: ([\\d.]+) \\([\\d\\-]+\\)"
```
Comment (Ash258) [inline bucket/dos2unix.json]: Do you need to be so strict?

```suggestion
        "regex": "Latest version: ([\\d.]+)\\s+\\(\\d"
```
Should be enough. Or?

File: bucket/dos2unix.json [hunk 2]
```diff
@@ -21,6 +21,16 @@
...
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://waterlan.home.xs4all.nl/dos2unix/dos2unix-$version-win64.zip"
```
Comment (Ash258) [inline bucket/dos2unix.json]: Why not sourceforge? Current URLs are from sourceforge

Comment (rasa) [reply bucket/dos2unix.json]: Understood, but the URLs are on https://waterlan.home.xs4all.nl/dos2unix.html, so let's consider these URLs are "authoritative". Also, Sourceforge returns 429 errors during many excavator runs, so minimizing hitting sourceforge will decease these 429s.

Comment (Ash258) [reply bucket/dos2unix.json]: Then run checkver to change current URLs

**Pull request merged**